### PR TITLE
fix build failure with empty qml bridges (closes #1033)

### LINKED ISSUE #1033: Bridge module without QObject type
### Discussed in https://github.com/KDAB/cxx-qt/discussions/1032

<div type='discussions-op-text'>

<sup>Originally posted by **SanderVocke** August 13, 2024</sup>
Hi there,

I'm finding that cxx-qt does not seem to allow defining a bridge module that does not have a `#[qobject]`-type in it. The reason seems to be that automoc is run on the resulting C++ files, and they error out when there are no Q_OBJECT types in the headers. For this reason, I am having to add dummy QObject types to all the bridge modules that I am making with e.g. some trivial types in them.

Am I missing something?

Example:

```rust
#[cxx_qt::bridge(cxx_file_stem = "shoop_rust_callable")]
pub mod ffi {
    unsafe extern "C++" {
        include!("cxx-qt-lib/qvariant.h");
        type QVariant = cxx_qt_lib::QVariant;
    }

    unsafe extern "C++" {
        include!("cxx-qt-shoop/shoop_rust_callable.h");
        type ShoopRustCallable = type_shoop_rust_callable::ShoopRustCallableRust;

        #[rust_name = "qvariant_can_convert_shoop_rust_callable"]
        fn qvariantCanConvertShoopRustCallable(variant: &QVariant) -> bool;

        #[rust_name = "register_metatype_shoop_rust_callable"]
        fn registerMetatypeShoopRustCallable(name : &mut String);
    }

    #[namespace = "rust::cxxqtlib1::qvariant"]
    unsafe extern "C++" {
        include!("cxx-qt-lib/qvariant.h");

        #[rust_name = "qvariant_construct_shoop_rust_callable"]
        fn qvariantConstruct(value: &ShoopRustCallable) -> QVariant;

        #[rust_name = "qvariant_value_or_default_shoop_rust_callable"]
        fn qvariantValueOrDefault(variant: &QVariant) -> ShoopRustCallable;
    }

    extern "RustQt" {
        #[qobject]
        type Dummy= super::DummyRust;
    }
}

use ffi::*;
#[derive(Default)]
pub struct DummyRust {}


(...)
```

This works, but fails as soon as the `Dummy` and `DummyRust` types are removed.</div>

### PR DESCRIPTION
**Pull request by aambrosano**
Closes #1033

### CODE REVIEW

File: examples/qml_features/rust/src/empty_bridge.rs [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#[cxx_qt::bridge]
```
Comment (ahayzen-kdab) [inline examples/qml_features/rust/src/empty_bridge.rs:1]: this is missing a SPDX header, copy one from elsewhere and update with your name etc :-)

**Comment by ahayzen-kdab** [comment]
Looks good thanks! Just some minor things to make CI happy

File: crates/qt-build-utils/src/lib.rs [hunk 2]
```diff
@@ -849,7 +849,26 @@ prefer :/qt/qml/{qml_uri_dirs}/
...
+        // Filter out empty jsons
+        let metatypes_json: Vec<_> = metatypes_json
+            .iter()
+            .filter(|f| {
+                std::fs::metadata(&f)
+                    .unwrap_or_else(|_| {
+                        panic!(
+                            "couldn't open json file {}",
+                            f.as_ref().to_string_lossy()
+                        )
```
[the remark this answers is not in the document]
Comment (aambrosano) [reply crates/qt-build-utils/src/lib.rs:862]: ach yeah, forgot it after the rebase

**Review by ahayzen-kdab — APPROVED** [review]
LGTM, thanks! :-)

**Pull request merged**